modelmesh-monitoring/base/servicemonitors/modelmesh-federated-metrics.yaml

```diff
@@ -0,0 +1,44 @@
+apiVersion: monitoring.coreos.com/v1
+kind: ServiceMonitor
+metadata:
+  name: modelmesh-federated-metrics
+  labels:
+    modelmesh-service: modelmesh-serving
+spec:
+  endpoints:
+    - interval: 30s
+      params:
+        'match[]':
+          - '{__name__= "haproxy_backend_http_average_response_latency_milliseconds"}'
+          - '{__name__= "haproxy_backend_http_responses_total"}'
+          - '{__name__= "container_cpu_usage_seconds_total"}'
+          - '{__name__= "container_memory_working_set_bytes"}'
+          - '{__name__= "node_namespace_pod_container:container_cpu_usage_seconds_total:sum_irate"}'
+          - '{__name__= "cluster:namespace:pod_cpu:active:kube_pod_container_resource_limits"}'
+          - '{__name__= "cluster:namespace:pod_cpu:active:kube_pod_container_resource_requests"}'
+          - '{__name__= "cluster:namespace:pod_memory:active:kube_pod_container_resource_requests"}'
+          - '{__name__= "cluster:namespace:pod_memory:active:kube_pod_container_resource_limits"}'
+          - '{__name__= "kube_persistentvolumeclaim_resource_requests_storage_bytes"}'
+          - '{__name__= "kubelet_volume_stats_used_bytes"}'
+          - '{__name__= "kubelet_volume_stats_capacity_bytes"}'
```

Comment:
This is for the benefit of the Dashboard. The plan is that the odh-dashboard will show some of these system metrics in the dashboard. We are federating these specific metrics from `openshift-monitoring` so that the dashboard does not have to look at multiple routes to get metrics.

Reply:
> The plan is that the odh-dashboard will show some of these system metrics in the dashboard
Isn't the dashboard for data-scientist ? SRE and apparented are not going to look at odh-dashboard, they'll use the standard one in their cluster (grafana, most likely)

Reply:
Some system metrics are also planned to be available in odh-dashboard even though SRE will continue to use other dashboards. This also comes from requirements around being able to export metrics into other services (like cloudwatch for example) which requires all metrics accessible by a single prometheus instance. We want to keep manifests for upstream and downstream in sync. I think that we should keep these federated metrics so that all metrics under the model-monitoring umbrella are available via the model-monitoring prometheus

Reply:
> all metrics under the model-monitoring umbrella are available via the model-monitoring prometheus

Well that's kinda my point. Those metrics are "owned" by (-> originate from if you want) k8s components, not by model-monitoring.

Reply:
Explained and resolved offline with @VannTen and @anishasthana